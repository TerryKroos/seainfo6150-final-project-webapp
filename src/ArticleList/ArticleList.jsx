import React from 'react';
import ArticleItem from "../ArticleItem/ArticleItem";
import styles from "./ArticleList.module.css";

const ArticleList = (props) => {
    
    return (
      <div>
        <ul className={styles.container}>
          {props.articles.map((article) => (
            <ArticleItem article={article} key={article.slug} />
          ))}
        </ul>
      </div>
    );
  };

  export default ArticleList;