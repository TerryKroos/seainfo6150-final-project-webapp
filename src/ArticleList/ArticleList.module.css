@import url('https://fonts.googleapis.com/css2?family=Inter:wght@200;700&display=swap');

  @media(min-width: 300px){
    .container {
      display: grid;
      grid-template-columns: repeat(1, 1fr);
      grid-auto-rows: 15em;
      gap:3em;
      margin-left: 7em;
      margin-right: 7em;
      background: #BBBBBB;
      list-style-type: none;
    }
  }
  
  @media (min-width: 600px){  
  .container {
      padding-top: 5em;
      display: grid;
      grid-template-columns: repeat(2, 1fr);
      grid-auto-rows: 40em;
      gap:10em;
      margin-left: 15em;
      margin-right: 15em;
      background: #BBBBBB;
      list-style-type: none;
      font-size: 5px;
      padding-bottom: 5em;
     

    }
  }
  @media screen and (min-width: 900px){
    .container {
      border-radius: 10px;
      display: grid;
      grid-template-columns: repeat(3, 1fr);
      grid-auto-rows: 50em;
      margin-left: 20em;
      margin-right: 20em;
      gap: 40px;
      list-style-type: none;
      padding: 15px;
      background-color: #BBBBBB;
      font-family: 'Inter', sans-serif;
      padding-bottom: 10em;
    }
  }
   
  
  
  