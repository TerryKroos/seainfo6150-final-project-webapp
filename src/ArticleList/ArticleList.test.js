
import React from "react";
import ArticleList from "./ArticleList";
import { render } from "@testing-library/react";

describe("ArticleList tests", () => {
    it("renders correctly", () => {
      const { container } = render(<ArticleList articles={[]}/>);
      expect(container).toMatchSnapshot();
    });
  
  });