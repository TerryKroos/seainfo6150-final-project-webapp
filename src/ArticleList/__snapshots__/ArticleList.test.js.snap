// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    />
  </div>
</div>
`;
